code/modules/mob/living/carbon/xenomorph/abilities/facehugger/facehugger_powers.dm

```diff
@@ -26,3 +26,13 @@
 			to_chat(owner, SPAN_WARNING("We cannot do that while squeezing and scuttling!"))
 			return FALSE
 	return ..()
+
+/datum/action/xeno_action/onclick/toggle_long_range/facehugger/use_ability()
+	. = ..()
+
+	var/mob/living/carbon/xenomorph/facehugger/facehugger = owner
+
+	if (facehugger.is_zoomed)
+		ADD_TRAIT(facehugger, TRAIT_IMMOBILIZED, TRAIT_SOURCE_ABILITY("Long-Range Sight"))
+	else
+		REMOVE_TRAIT(facehugger, TRAIT_IMMOBILIZED, TRAIT_SOURCE_ABILITY("Long-Range Sight"))
```

Comment:
Two things:
- There are more ways for the ability to be toggled off than clicking the button (e.g. resting). You need to instead hook this logic instead into `/mob/living/carbon/xenomorph/proc/zoom_in()` and `/mob/living/carbon/xenomorph/proc/zoom_out()`
- TRAIT_IMMOBILIZED does not currently prevent pouncing - so you can just pouce to move around.